==13736== Memcheck, a memory error detector
==13736== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==13736== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==13736== Command: /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global
==13736== 
==13736== Conditional jump or move depends on uninitialised value(s)
==13736==    at 0x68BDAD: wasmparser::validator::Validator::end (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6818AC: wasmparser::validator::Validator::payload (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x680B21: wasmparser::validator::Validator::validate_all (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x299FEC: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x265688: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::validate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF34: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==  Uninitialised value was created by a stack allocation
==13736==    at 0x299F00: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 
==13736== HEAP SUMMARY:
==13736==     in use at exit: 32,374 bytes in 116 blocks
==13736==   total heap usage: 846 allocs, 730 frees, 147,780 bytes allocated
==13736== 
==13736== 4 bytes in 1 blocks are still reachable in loss record 1 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x6622FE: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x659F11: wasmer_compiler::translator::sections::parse_global_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6671C5: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 7 bytes in 2 blocks are still reachable in loss record 2 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x65DF3F: wasmer_compiler::translator::environ::ModuleEnvironment::declare_global_export (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65A9AA: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x66713D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 8 bytes in 1 blocks are still reachable in loss record 3 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x662F11: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x658C24: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x666ECF: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 8 bytes in 1 blocks are still reachable in loss record 4 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x662F11: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x658BCF: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x666ECF: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 8 bytes in 1 blocks are still reachable in loss record 5 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2E473A: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2A0481: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x284AA8: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 12 bytes in 1 blocks are still reachable in loss record 6 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x662120: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x659716: wasmer_compiler::translator::sections::parse_function_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x666BB4: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 16 bytes in 1 blocks are still reachable in loss record 7 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x6BFCA8: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x6C44E6: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6CC630: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6BF1F5: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x58346D: wasmer_engine::trap::error::RuntimeError::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CAE0: wasmer::externals::global::Global::set (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25C66D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 16 bytes in 1 blocks are indirectly lost in loss record 8 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x260E4D: wasmer_engine::tunables::Tunables::create_globals (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x280C71: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57D2ED: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x582AE2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BF3D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 16 bytes in 1 blocks are indirectly lost in loss record 9 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x263936: <alloc::boxed::Box<[T]> as core::clone::Clone>::clone (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x280EB7: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57D2ED: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x582AE2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BF3D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 23 bytes in 3 blocks are still reachable in loss record 10 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x65DC9F: wasmer_compiler::translator::environ::ModuleEnvironment::declare_func_export (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65A96B: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x66713D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 24 bytes in 1 blocks are indirectly lost in loss record 11 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x263A26: <alloc::boxed::Box<[T]> as core::clone::Clone>::clone (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x280E8A: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57D2ED: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x582AE2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BF3D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 32 bytes in 1 blocks are still reachable in loss record 12 of 52
==13736==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==13736==    by 0x4E3F165: dlsym (dlsym.c:70)
==13736==    by 0x73C4B3: fetch (weak.rs:61)
==13736==    by 0x73C4B3: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==13736==    by 0x73C4B3: min_stack_size (thread.rs:449)
==13736==    by 0x73C4B3: std::sys::unix::thread::Thread::new (thread.rs:49)
==13736==    by 0x2DD7D6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0B56: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E170F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 32 bytes in 1 blocks are still reachable in loss record 13 of 52
==13736==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==13736==    by 0x2EF40D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E6EE2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DE87B: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2429: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E053A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2FD0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DD99E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x73C829: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13736==    by 0x73C829: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13736==    by 0x73C829: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13736==    by 0x52516DA: start_thread (pthread_create.c:463)
==13736==    by 0x57A271E: clone (clone.S:95)
==13736== 
==13736== 32 bytes in 2 blocks are indirectly lost in loss record 14 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x6AF7B9: wasmer_vm::global::Global::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x260EF8: wasmer_engine::tunables::Tunables::create_globals (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x280C71: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57D2ED: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x582AE2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BF3D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 40 bytes in 1 blocks are still reachable in loss record 15 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x6CBC8B: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6BFC98: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x6C44E6: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6CC630: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6BF1F5: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x58346D: wasmer_engine::trap::error::RuntimeError::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CAE0: wasmer::externals::global::Global::set (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25C66D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 64 bytes in 4 blocks are still reachable in loss record 16 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2E6D3C: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E7BB7: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0FE6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 64 bytes in 4 blocks are still reachable in loss record 17 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x73C40C: alloc (alloc.rs:80)
==13736==    by 0x73C40C: alloc (alloc.rs:174)
==13736==    by 0x73C40C: exchange_malloc (alloc.rs:277)
==13736==    by 0x73C40C: std::sys::unix::thread::Thread::new (thread.rs:44)
==13736==    by 0x2DD7D6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0B56: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E170F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 72 bytes in 1 blocks are still reachable in loss record 18 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x6435ED: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x64725B: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x286B65: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x280E45: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57D2ED: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x582AE2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BF3D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 80 bytes in 2 blocks are indirectly lost in loss record 19 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x6A5D9B: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6AF7D5: wasmer_vm::global::Global::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x260EF8: wasmer_engine::tunables::Tunables::create_globals (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x280C71: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57D2ED: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x582AE2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BF3D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 88 bytes in 1 blocks are possibly lost in loss record 20 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x655EF2: hashbrown::raw::inner::RawTable<T>::fallible_with_capacity (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65590E: _ZN9hashbrown3raw5inner17RawTable$LT$T$GT$14reserve_rehash17he29be5fc862d84d8E.llvm.8588116468508767673 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65AAA1: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x66713D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 96 bytes in 1 blocks are still reachable in loss record 21 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x662027: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6589D7: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x666ECF: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 96 bytes in 1 blocks are still reachable in loss record 22 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x66282C: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65A523: wasmer_compiler::translator::sections::parse_global_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x6671C5: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 96 bytes in 3 blocks are still reachable in loss record 23 of 52
==13736==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==13736==    by 0x2EF40D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E6EE2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DE7E6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2429: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E053A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2FD0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DD99E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x73C829: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13736==    by 0x73C829: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13736==    by 0x73C829: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13736==    by 0x52516DA: start_thread (pthread_create.c:463)
==13736==    by 0x57A271E: clone (clone.S:95)
==13736== 
==13736== 96 bytes in 2 blocks are indirectly lost in loss record 24 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x260F05: wasmer_engine::tunables::Tunables::create_globals (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x280C71: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57D2ED: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x582AE2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BF3D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 128 bytes in 4 blocks are still reachable in loss record 25 of 52
==13736==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==13736==    by 0x708A3E: register_dtor (fast_thread_local.rs:33)
==13736==    by 0x708A3E: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==13736==    by 0x708A3E: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==13736==    by 0x72B239: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==13736==    by 0x72B239: __getit (local.rs:175)
==13736==    by 0x72B239: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==13736==    by 0x72B239: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==13736==    by 0x72B239: std::sys_common::thread_info::set (thread_info.rs:40)
==13736==    by 0x2DD95C: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x73C829: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13736==    by 0x73C829: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13736==    by 0x73C829: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13736==    by 0x52516DA: start_thread (pthread_create.c:463)
==13736==    by 0x57A271E: clone (clone.S:95)
==13736== 
==13736== 160 bytes in 4 blocks are still reachable in loss record 26 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2E49CC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E202B: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E7E18: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2F60: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E108A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 160 bytes in 4 blocks are still reachable in loss record 27 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2E49CC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2056: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E7E18: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2F60: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E108A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 160 bytes in 4 blocks are still reachable in loss record 28 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2E49CC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E7FDB: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E1100: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 160 bytes in 4 blocks are still reachable in loss record 29 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x709DDE: alloc (alloc.rs:80)
==13736==    by 0x709DDE: alloc (alloc.rs:174)
==13736==    by 0x709DDE: exchange_malloc (alloc.rs:277)
==13736==    by 0x709DDE: new<()> (mutex.rs:223)
==13736==    by 0x709DDE: std::thread::Thread::new (mod.rs:1149)
==13736==    by 0x2DD6C1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0B56: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E170F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 160 bytes in 4 blocks are still reachable in loss record 30 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2DD6E3: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0B56: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E170F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 192 bytes in 4 blocks are still reachable in loss record 31 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x726ED6: alloc (alloc.rs:80)
==13736==    by 0x726ED6: alloc (alloc.rs:174)
==13736==    by 0x726ED6: exchange_malloc (alloc.rs:277)
==13736==    by 0x726ED6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13736==    by 0x2E2036: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E7E18: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2F60: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E108A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 192 bytes in 4 blocks are still reachable in loss record 32 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x726ED6: alloc (alloc.rs:80)
==13736==    by 0x726ED6: alloc (alloc.rs:174)
==13736==    by 0x726ED6: exchange_malloc (alloc.rs:277)
==13736==    by 0x726ED6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13736==    by 0x2E2061: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E7E18: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2F60: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E108A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 192 bytes in 4 blocks are still reachable in loss record 33 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x726ED6: alloc (alloc.rs:80)
==13736==    by 0x726ED6: alloc (alloc.rs:174)
==13736==    by 0x726ED6: exchange_malloc (alloc.rs:277)
==13736==    by 0x726ED6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13736==    by 0x2E7FE1: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E1100: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 192 bytes in 4 blocks are still reachable in loss record 34 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x709E42: alloc (alloc.rs:80)
==13736==    by 0x709E42: alloc (alloc.rs:174)
==13736==    by 0x709E42: exchange_malloc (alloc.rs:277)
==13736==    by 0x709E42: new (condvar.rs:130)
==13736==    by 0x709E42: std::thread::Thread::new (mod.rs:1150)
==13736==    by 0x2DD6C1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0B56: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E170F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 280 bytes in 1 blocks are still reachable in loss record 35 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x661F27: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65A7F3: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x66713D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x65D0E8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2845C4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736== 
==13736== 320 bytes in 4 blocks are still reachable in loss record 36 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x709E91: alloc (alloc.rs:80)
==13736==    by 0x709E91: alloc (alloc.rs:174)
==13736==    by 0x709E91: exchange_malloc (alloc.rs:277)
==13736==    by 0x709E91: new<std::thread::Inner> (sync.rs:324)
==13736==    by 0x709E91: std::thread::Thread::new (mod.rs:1145)
==13736==    by 0x2DD6C1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0B56: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E170F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 384 bytes in 1 blocks are still reachable in loss record 37 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2E61EC: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2F1C: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E108A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 416 bytes in 4 blocks are still reachable in loss record 38 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2DD778: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0B56: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E170F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 512 bytes in 1 blocks are still reachable in loss record 39 of 52
==13736==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x72C949: aligned_malloc (alloc.rs:95)
==13736==    by 0x72C949: alloc (alloc.rs:22)
==13736==    by 0x72C949: __rdl_alloc (alloc.rs:304)
==13736==    by 0x2E6506: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E10BA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 640 bytes in 1 blocks are still reachable in loss record 40 of 52
==13736==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x72C949: aligned_malloc (alloc.rs:95)
==13736==    by 0x72C949: alloc (alloc.rs:22)
==13736==    by 0x72C949: __rdl_alloc (alloc.rs:304)
==13736==    by 0x2E12F6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 640 bytes in 1 blocks are still reachable in loss record 41 of 52
==13736==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x72C949: aligned_malloc (alloc.rs:95)
==13736==    by 0x72C949: alloc (alloc.rs:22)
==13736==    by 0x72C949: __rdl_alloc (alloc.rs:304)
==13736==    by 0x2EF812: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2EF4AD: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E6EE2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DE7E6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2429: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E053A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2FD0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DD99E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 664 bytes in 1 blocks are still reachable in loss record 42 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x284933: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x26583A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57CF8F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25EE61: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BE13: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 696 (432 direct, 264 indirect) bytes in 1 blocks are definitely lost in loss record 43 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x6AB07D: wasmer_vm::instance::InstanceHandle::allocate_instance (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x280811: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57D2ED: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x582AE2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BF3D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 808 bytes in 1 blocks are still reachable in loss record 44 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x64EB4F: alloc::collections::btree::map::BTreeMap<K,V>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x647661: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x286B65: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x280E45: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x57D2ED: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x582AE2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25BF3D: exported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x25DA72: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x72DEE2: {{closure}} (rt.rs:52)
==13736==    by 0x72DEE2: do_call<closure-0,i32> (panicking.rs:348)
==13736==    by 0x72DEE2: try<i32,closure-0> (panicking.rs:325)
==13736==    by 0x72DEE2: catch_unwind<closure-0,i32> (panic.rs:394)
==13736==    by 0x72DEE2: std::rt::lang_start_internal (rt.rs:51)
==13736==    by 0x25D671: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 1,152 bytes in 4 blocks are possibly lost in loss record 45 of 52
==13736==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==13736==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==13736==    by 0x5252227: allocate_stack (allocatestack.c:627)
==13736==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==13736==    by 0x73C55C: std::sys::unix::thread::Thread::new (thread.rs:66)
==13736==    by 0x2DD7D6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0B56: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E170F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 1,520 bytes in 1 blocks are still reachable in loss record 46 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2E1152: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2A0481: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x284AA8: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 1,536 bytes in 4 blocks are still reachable in loss record 47 of 52
==13736==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x72C949: aligned_malloc (alloc.rs:95)
==13736==    by 0x72C949: alloc (alloc.rs:22)
==13736==    by 0x72C949: __rdl_alloc (alloc.rs:304)
==13736==    by 0x2E6DE0: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E7BB7: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0FE6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 2,008 bytes in 1 blocks are still reachable in loss record 48 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2EEF5E: crossbeam_epoch::sync::queue::Queue<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2EF7E2: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2EF4AD: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E6EE2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DE7E6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2429: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E053A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2FD0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DD99E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x73C829: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13736==    by 0x73C829: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13736==    by 0x73C829: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13736== 
==13736== 2,040 bytes in 1 blocks are still reachable in loss record 49 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2EC83F: crossbeam_epoch::internal::Local::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2EF434: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E6EE2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DE87B: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2429: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E053A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2FD0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DD99E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x73C829: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13736==    by 0x73C829: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13736==    by 0x73C829: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13736==    by 0x52516DA: start_thread (pthread_create.c:463)
==13736==    by 0x57A271E: clone (clone.S:95)
==13736== 
==13736== 4,096 bytes in 4 blocks are still reachable in loss record 50 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2E6D20: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E7BB7: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E0FE6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E4706: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x727589: std::sync::once::Once::call_inner (once.rs:416)
==13736==    by 0x2E0CD0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E3ACB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x28CCB1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x293855: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x298008: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2B3A51: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736== 
==13736== 6,080 bytes in 4 blocks are still reachable in loss record 51 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2DFF08: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2FD0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DD99E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x73C829: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13736==    by 0x73C829: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13736==    by 0x73C829: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13736==    by 0x52516DA: start_thread (pthread_create.c:463)
==13736==    by 0x57A271E: clone (clone.S:95)
==13736== 
==13736== 6,120 bytes in 3 blocks are still reachable in loss record 52 of 52
==13736==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13736==    by 0x2EC83F: crossbeam_epoch::internal::Local::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2EF434: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E6EE2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DE7E6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2429: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E053A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2E2FD0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x2DD99E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/exported-global)
==13736==    by 0x73C829: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13736==    by 0x73C829: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13736==    by 0x73C829: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13736==    by 0x52516DA: start_thread (pthread_create.c:463)
==13736==    by 0x57A271E: clone (clone.S:95)
==13736== 
==13736== LEAK SUMMARY:
==13736==    definitely lost: 432 bytes in 1 blocks
==13736==    indirectly lost: 264 bytes in 9 blocks
==13736==      possibly lost: 1,240 bytes in 5 blocks
==13736==    still reachable: 30,438 bytes in 101 blocks
==13736==         suppressed: 0 bytes in 0 blocks
==13736== 
==13736== For counts of detected and suppressed errors, rerun with: -v
==13736== ERROR SUMMARY: 4 errors from 4 contexts (suppressed: 0 from 0)
